pkg/kf/commands/group/group.go

```diff
@@ -0,0 +1,160 @@
+	Name     string
+	Commands []*cobra.Command
+}
+
+// CommandGroups is a list of CommandGroups.
+type CommandGroups []CommandGroup
+
+// defaultTemplateFuncs are the func cobra provides its users.
+var defaultTemplateFuncs = template.FuncMap{
+	"trim":                    strings.TrimSpace,
+	"trimRightSpace":          trimRightSpace,
+	"trimTrailingWhitespaces": trimRightSpace,
+	"rpad":                    rpad,
+	"gt":                      cobra.Gt,
+	"eq":                      cobra.Eq,
+}
+
+// PrintTemplate prints the provided template to the output writer.
+func PrintTemplate(out io.Writer, text string, data interface{}, templateFuncs *template.FuncMap) error {
+	t := template.New("")
+	if templateFuncs == nil {
+		t.Funcs(defaultTemplateFuncs)
+	} else {
+		t.Funcs(*templateFuncs)
+	}
+
+	template.Must(t.Parse(text))
+	return t.Execute(out, data)
+}
+
+// rpad adds padding to the right of a string.
+func rpad(s string, padding int) string {
+	template := fmt.Sprintf("%%-%ds", padding)
+	return fmt.Sprintf(template, s)
+}
+
+/// trimRightSpace trims tailing whitespace.
+func trimRightSpace(s string) string {
+	return strings.TrimRightFunc(s, unicode.IsSpace)
+}
+
+// CalculateMinWidth gets the minimum width required for all command names.
+func CalculateMinWidth(groups CommandGroups) int {
+	minWidth := 0
+	for _, group := range groups {
+		for _, c := range group.Commands {
+			if c == nil {
+				continue
+			}
+			if len(c.Name()) > minWidth {
+				minWidth = len(c.Name())
+			}
+		}
+	}
+	return minWidth
+}
+
+// PrintTrimmedMultilineString does just that.
+func PrintTrimmedMultilineString(str string, out io.Writer) {
+	if str == "" {
+		return
+	}
+
+	for _, line := range strings.Split(str, "\n") {
+		fmt.Fprintln(out, strings.TrimSpace(line))
+	}
+}
+
+// CommandGroupUsageFunc returns a UsageFunc a root level command can use.
+func CommandGroupUsageFunc(groups CommandGroups, templateFuncs *template.FuncMap) func(*cobra.Command) error {
+	return func(command *cobra.Command) error {
+		out := command.OutOrStdout()
+		fmt.Fprintln(out)
+		return PrintTemplate(out, command.UsageTemplate(), command, templateFuncs)
+	}
+}
+
+// CommandGroupHelpFunc returns a HelpFunc a root level command can use.
+func CommandGroupHelpFunc(rootCommand *cobra.Command, groups CommandGroups, templateFuncs *template.FuncMap) func(*cobra.Command, []string) {
+	return func(command *cobra.Command, args []string) {
+		out := command.OutOrStdout()
+
+		// not the root level, use the default template
+		if rootCommand != command {
+			err := PrintTemplate(out, command.HelpTemplate(), command, templateFuncs)
+			if err != nil {
+				panic(fmt.Sprintf("Error printing help: %v", err))
+			}
+			return
+		}
+
+		PrintTrimmedMultilineString(command.Long, out)
+		fmt.Fprintln(out)
+
+		minWidth := CalculateMinWidth(groups)
+		// add 2 for the prefix spaces, 1 for the padding between cols
+		minWidth += 3
+
+		tabout := tabwriter.NewWriter(out, minWidth, 8, 1, ' ', 0)
```

Comment:
instead of using a tabwriter, would it make sense to just `rpad` out the string?

Reply:
good call!